staging/src/k8s.io/kubectl/pkg/validation/validation.go

```diff
@@ -29,17 +30,33 @@ import (
 
 // schemaValidation validates the object against an OpenAPI schema.
 type schemaValidation struct {
+	resourcesGetter openapi.OpenAPIResourcesGetter
+
+	// Cached OpenAPI V2 resources specs
+	sync.Once
 	resources openapi.Resources
+	err       error
 }
 
 // NewSchemaValidation creates a new Schema that can be used
 // to validate objects.
-func NewSchemaValidation(resources openapi.Resources) Schema {
+func NewSchemaValidation(resourcesGetter openapi.OpenAPIResourcesGetter) Schema {
 	return &schemaValidation{
-		resources: resources,
+		resourcesGetter: resourcesGetter,
 	}
 }
 
+// retrieveResources gets (once) and caches the openapi.Resources, and
+// the error condition for the retrieval.
+func (v *schemaValidation) retrieveResources() (openapi.Resources, error) {
```

Comment:
The code in the factory is not great, but I think this is actually already done in the factory, no?
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kubectl/pkg/util/openapi/openapi_getter.go#L56-L63

Reply:
Yes. You're right. We don't need to cache the `Resources`, since they're cached already in the `OpenAPIParser` (which is used for the `OpenAPISchema()` call in the factory. We can remove this, and do a simple call on the `OpenAPIResourcesGetter#OpenAPISchema()`.

Reply:
...the entire lazy-load mechanism is dependent on the implementation of the `OpenAPISchema()` within `factory_client_access.go`.